### PR TITLE
Issue #271 - Allow missing values when an `sdmxa:obsStatus` Attribute explains why it's missing.

### PR DESCRIPTION
**Pull request by robons**
Issues raised as a result of this PR:

* #273
* #274

Satisfies Issue #271.

### CODE REVIEW

File: csvcubed/csvcubed/utils/qb/cube.py [hunk 1]
```diff
@@ -151,9 +164,73 @@ def _validate_observation_value_constraints(cube: Cube) -> List[ValidationError]
             errors += _validate_observation_value(obs_val_column, multi_units_columns)
             errors += _validate_single_measure_cube(cube, obs_val_column)
 
+        errors += _validate_missing_observation_values(cube, observed_value_columns[0])
+
     return errors
 
 
+def _validate_missing_observation_values(
```
Comment (canwaf) [inline csvcubed/csvcubed/utils/qb/cube.py:172]: I think that the observations should be in their own module.

Comment (robons) [reply csvcubed/csvcubed/utils/qb/cube.py:172]: I think at some point we'll end up splitting out all of the different validations into separate files. What makes you suggest it now?

File: csvcubed/tests/behaviour/qbwriter.feature [hunk 2]
```diff
@@ -427,4 +427,26 @@ Feature: Test outputting CSV-Ws with Qb flavouring.
...
+    When the cube is serialised to CSV-W
+    Then csvlint validation of "good-qube.csv-metadata.json" should succeed
+
+  Scenario: Observation Values are Required where an `sdmxa:ObsStatus` Attribute Column is present but no value is set.
+    Given a single-measure QbCube named "Bad Qube" with missing observation values and missing `sdmxa:obsStatus` replacements
+    Then the CSVqb should fail validation with "Missing value(s) found for 'Value' in row(s) 0"
+    When the cube is serialised to CSV-W
+    # Unfortunately, CSV-W validation will *not* catch this error since the obs column cannot be marked as `required`
+    # since an `sdmxa:obsStatus` Attribute column has been defined.
+    Then csvlint validation of "bad-qube.csv-metadata.json" should succeed
```
Comment (canwaf) [inline csvcubed/tests/behaviour/qbwriter.feature:451]: Is there a thing to do with csvcheck that could solve this?

File: csvcubed/csvcubed/utils/qb/cube.py (hunk 1 above)
Comment (canwaf) [reply csvcubed/csvcubed/utils/qb/cube.py:172]: It's grown this module by 50% LoC, which is some artibtrary mental threshold for some minor refactoring. Won't hurt if you don't want to do it now, but it is easily done now.

**Review by canwaf — APPROVED** [review]
I think pulling the observation stuff out of the cube.py is worthwhile now; however not a dealbreaker.

File: csvcubed/tests/behaviour/qbwriter.feature (hunk 2 above)
Comment (robons) [reply csvcubed/tests/behaviour/qbwriter.feature:451]: I don't think csvw-check is the right place to solve the issue. We'd effectively be making an unofficial extension to their CSV-W spec.

However, there is definitely scope for more CSV-qb validation which I believe is in our issues long-term.

**Pull request merged**